osu.Game.Rulesets.Osu/Utils/OsuHitObjectGenerationUtils.cs

```diff
@@ -146,5 +146,40 @@ public static void ReflectVertically(OsuHitObject osuObject)
 
             slider.Path = new SliderPath(controlPoints, slider.Path.ExpectedDistance.Value);
         }
+
+        /// <summary>
+        /// Rotate a slider about its start position by the specified angle.
+        /// </summary>
+        /// <param name="slider">The slider to be rotated.</param>
+        /// <param name="rotation">The angle to rotate the slider by.</param>
+        public static void RotateSlider(Slider slider, float rotation)
+        {
+            void rotateNestedObject(OsuHitObject nested) => nested.Position = rotateVector(nested.Position - slider.Position, rotation) + slider.Position;
+
+            slider.NestedHitObjects.OfType<SliderTick>().ForEach(rotateNestedObject);
+            slider.NestedHitObjects.OfType<SliderRepeat>().ForEach(rotateNestedObject);
+
+            var controlPoints = slider.Path.ControlPoints.Select(p => new PathControlPoint(p.Position, p.Type)).ToArray();
+            foreach (var point in controlPoints)
+                point.Position = rotateVector(point.Position, rotation);
+
+            slider.Path = new SliderPath(controlPoints, slider.Path.ExpectedDistance.Value);
+        }
+
+        /// <summary>
+        /// Rotate a vector by the specified angle.
+        /// </summary>
+        /// <param name="vector">The vector to be rotated.</param>
+        /// <param name="rotation">The angle to rotate the vector by.</param>
+        /// <returns>The rotated vector.</returns>
+        private static Vector2 rotateVector(Vector2 vector, float rotation)
```

Comment:
According to Wikipedia a vector is "a geometric object that has magnitude (or length) and direction" (which, in 2-dimensional euclidean space, can also be represented as a set of two scalar components `x` and `y`). A vector doesn't have a position because it's not an actual arrow on a plane or anything, it is the position of one point relative to another one.
And for my personal opinion on the terms used, I think this method does what you would expect from it. Although I think it would be a good idea to clarify that the rotation must be in radian, not degrees, because when taking an angle as an input from the user, it will probably be in degrees.

Reply:
if we're quoting wikipedia... a bit further down, in the same article:

> Being an arrow, a Euclidean vector possesses a definite initial point and terminal point. A vector with fixed initial and terminal point is called a bound vector. When only the magnitude and direction of the vector matter, then the particular initial point is of no importance, and the vector is called a free vector.

Reply:
Yes, bound vectors also exist, but that doesn't really matter here. Free vectors only have a magnitude and direction, and since a `Vector2` doesn't have a position, only the magnitude and direction matter, which makes it a free vector.

Reply:
> the point being, "rotate a point around the euclidean origin" is a lot less ambiguous and easier to explain than "rotate a vector given by a pair of coordinates".

I still prefer my original wording over your suggestion, since `Vector2` represents a free vector, and rotating a free vector is an unambiguous action. It is more concise than explaining how the `Vector2` represents a point IMO.

However, do feel free to push changes if you have strong opinions over how this should be worded.